R/layer-sf.R

```diff
@@ -35,6 +35,21 @@ LayerSf <- ggproto("LayerSf", Layer,
         self$mapping$geometry <- as.name(geometry_col)
       }
     }
+
+    # automatically determine the legend type
+    if (is.na(self$show.legend) || isTRUE(self$show.legend)) {
+      if (is_sf(data)) {
+        sf_type <- detect_sf_type(data)
+        if (sf_type == "point")
+          self$geom_params$legend <- "point"
+        else if (sf_type == "line")
+          self$geom_params$legend <- "line"
+        else self$geom_params$legend <- "polygon"
```

Comment:
Minor comment. According to [the tidyverse style guide](https://style.tidyverse.org/syntax.html#inline-statements), 

> It’s ok to drop the curly braces for very simple statements that fit on one line, as long as they don’t have side-effects.

so you need `{}` here since they are multiple lines.